netComparator/cmd/main.go

```diff
@@ -0,0 +1,426 @@
+package main
+
+import (
+	"context"
+	"embed"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/avast/retry-go"
+	"github.com/multiversx/mx-sdk-go/blockchain"
+	"github.com/multiversx/mx-sdk-go/core"
+	"github.com/urfave/cli"
+	"html/template"
+	"net/http"
+	"os"
+	"reflect"
+	"strconv"
+	"sync"
+	"time"
+)
+
+const (
+	apiURL = "https://api.multiversx.com"
+	shfURL = "https://express-api-shadowfork-four.elrond.ro"
+)
+
+type transactions struct {
+	TxHash string `json:"txHash"`
+}
+
+type Transaction struct {
+	TxHash         string `json:"txHash"`
+	GasLimit       int    `json:"gasLimit"`
+	GasPrice       int    `json:"gasPrice"`
+	GasUsed        int    `json:"gasUsed"`
+	MiniBlockHash  string `json:"miniBlockHash"`
+	Nonce          int    `json:"nonce"`
+	Receiver       string `json:"receiver"`
+	ReceiverAssets struct {
+		Name string   `json:"name"`
+		Tags []string `json:"tags"`
+	} `json:"receiverAssets"`
+	ReceiverShard int    `json:"receiverShard"`
+	Round         int    `json:"round"`
+	Sender        string `json:"sender"`
+	SenderShard   int    `json:"senderShard"`
+	Signature     string `json:"signature"`
+	Status        string `json:"status"`
+	Value         string `json:"value"`
+	Fee           string `json:"fee"`
+	Timestamp     int    `json:"timestamp"`
+	Data          string `json:"data"`
+	Function      string `json:"function"`
+	Action        struct {
+		Category string `json:"category"`
+		Name     string `json:"name"`
+	} `json:"action"`
+	Results []struct {
+		Hash         string `json:"hash"`
+		Timestamp    int    `json:"timestamp"`
+		Nonce        int    `json:"nonce"`
+		GasLimit     int    `json:"gasLimit"`
+		GasPrice     int    `json:"gasPrice"`
+		Value        string `json:"value"`
+		Sender       string `json:"sender"`
+		Receiver     string `json:"receiver"`
+		SenderAssets struct {
+			Name string   `json:"name"`
+			Tags []string `json:"tags"`
+		} `json:"senderAssets"`
+		Data           string `json:"data"`
+		PrevTxHash     string `json:"prevTxHash"`
+		OriginalTxHash string `json:"originalTxHash"`
+		CallType       string `json:"callType"`
+		MiniBlockHash  string `json:"miniBlockHash"`
+		Logs           struct {
+			Id      string `json:"id"`
+			Address string `json:"address"`
+			Events  []struct {
+				Identifier string   `json:"identifier"`
+				Address    string   `json:"address"`
+				Topics     []string `json:"topics"`
+				Order      int      `json:"order"`
+				Data       string   `json:"data,omitempty"`
+			} `json:"events"`
+		} `json:"logs"`
+	} `json:"results"`
+	Price float64 `json:"price"`
+	Logs  struct {
+		Id            string `json:"id"`
+		Address       string `json:"address"`
+		AddressAssets struct {
+			Name string   `json:"name"`
```

Comment:
this is redefined in several places, you could separately define it once and re-use it in all places.